daffodil-lib/src/main/scala/org/apache/daffodil/lib/util/StackLike.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.lib.util
+
+import scala.collection.mutable.ListBuffer
+
+/**
+ * Compatibility class for 2.12 and 2.13 since ArrayStack and Stack
+ * have been deprecated in 2.13. This allows us to maintain the same
+ * functionality as stack while using ListBuffer instead
+ *
+ * ArrayDeque is the recommended replacement but is not available in 2.12
```

Comment:
Drop comment. 

Even if ArrayDeque is recommended, there is value in calling a stack a Stack. I would not recommend removing this even if ArrayDeque has all the functionality, because well, an ArrayDeque presumably also has non-stack additional behavior and if we are only using the Stack behavior then that is worth knowing, preferably by having this class/trait that enforces this. 

I mean if I am reading code, and it uses a data structure that uses "double ended queue", then I'm assuming they chose that because they need more than just stack behavior. So it's misleading to me if someone only needs  stack behavior, but uses a more general data structure which does not have 'stack' in its name.